### PR TITLE
Move HMAC to EVP_MAC

### PR DESCRIPTION
**Pull request by levitte**
Add HMAC implementation 
Integrate HMAC EVP_PKEY_METHOD into generic MAC EVP_PKEY_METHOD

This PR sits on top of #7393, and also works as a proof of concept.

### CODE REVIEW

**Comment by levitte** [comment]
Updated and documentation added.

File: doc/man7/EVP_MAC_HMAC.pod [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+There are no corresponding string control types.
+
+=item B<EVP_MAC_CTRL_SET_ENGINE>
+
+=item B<EVP_MAC_CTRL_SET_MD>
+
+These work as described in L<EVP_MAC(3)/CONTROLS>.
+
+There are no corresponding string control types.
```
Comment (levitte) [inline doc/man7/EVP_MAC_HMAC.pod]: So hmmm, the CMAC implementation does have a "cipher" string, so this should have a corresponding "digest" string, to be consistent.

**Comment by levitte** [comment]
Now that #7393 is in, I rebased this on top of master.

**Review by paulidale — APPROVED** [review]

**Comment by levitte** [comment]
Thanks for the review!

Merged.

388de53c274dee20c07eee7ff892108668fb3a61 Make sure at least one HMAC test still uses the EVP_PKEY method
f8c9a8e325b23f4b3de67e9a0d385355f81bd6fc EVP_MAC: Integrate HMAC EVP_PKEY_METHOD into generic MAC EVP_PKEY_METHOD
6723f86746ab7e8ff9a914603db4f85c53eafc7c EVP_MAC: Add HMAC implementation

**Pull request closed without merging**